extensions/typescript-language-features/src/commands.ts

```diff
@@ -105,6 +105,18 @@ export class JavaScriptGoToProjectConfigCommand implements Command {
 	}
 }
 
+export class ConfigurePluginCommand implements Command {
+	public readonly id = '_typescript.configurePlugin';
+
+	public constructor(
+		private readonly lazyClientHost: Lazy<TypeScriptServiceClientHost>,
+	) { }
+
+	public execute(pluginName: string, configuration: any) {
+		this.lazyClientHost.value.serviceClient.configurePlugin(pluginName, configuration, true /* reconfigureOnRestart */);
```

Comment:
Accessing `lazyClientHost.value` will force the typescript server to be created which can be expensive. Let's try to defer this instead so that if the server has not been activated, this command just stores received config values until activation takes place.

Reply:
One option: have a `PluginConfigurationProvider` class that stores the plugin config map. Then both the command and the service client share this object.

The class interface could be something like:

```ts
class PluginConfigurationProvider {
    set(name: string, config: any): void;
    
    readonly entries: Iterator<{ plugin: string, config: any }>;
 
    readonly onChangedConfiguration: vscode.Event<{ plugin: string, config: any }>;
} 
```

Reply:
OK, so what's a reliable way to check whether the service is activated yet so I know whether to send the command yet? Check `lazyClientHost.hasValue`?

Reply:
Yes. I'd try to avoid accessing the host entirely though and just go through the `PluginConfigurationProvider` object. Then the service can listen to the `onChangedConfiguration` to pick up changes that are broadcast by the implementation of `set`